pkg/uroot/uroot.go

```diff
@@ -63,28 +63,35 @@ var DefaultRamfs = []cpio.Record{
 	cpio.StaticFile("etc/localtime", gmt0, 0644),
 }
 
-// Opts are the arguments to CreateInitramfs.
-type Opts struct {
-	// Env is the build environment (OS, arch, etc).
-	Env golang.Environ
-
+// Commands specifies a list of packages to build with a specific builder.
+type Commands struct {
 	// Builder is the build format.
-	//
-	// This can currently be "source" or "bb".
 	Builder Build
 
-	// Archiver is the initramfs archival format.
-	//
-	// Only "cpio" is currently supported.
-	Archiver Archiver
-
 	// Packages are the Go packages to add to the archive.
 	//
 	// Currently allowed formats:
 	//   Go package imports; e.g. github.com/u-root/u-root/cmds/ls
 	//   Paths to Go package directories; e.g. $GOPATH/src/github.com/u-root/u-root/cmds/ls
 	//   Globs of paths to Go package directories; e.g. ./cmds/*
 	Packages []string
+}
+
+// Opts are the arguments to CreateInitramfs.
+type Opts struct {
+	// Env is the build environment (OS, arch, etc).
+	Env golang.Environ
+
+	// Commands specify packages to build using a specific builder.
+	Commands []Commands
```

Comment:
A mixed build would look something like this:

```go
opts := uroot.Opts{
  Commands: []uroot.Commands{
    {
       Builder: uroot.BBBuild,
       Packages: []string{
         "github.com/u-root/u-root/cmds/init",
         "github.com/u-root/u-root/cmds/ls",
       },
    },
    {
       Builder: uroot.BinaryBuild,
       Packages: []string{
         "github.com/u-root/u-root/cmds/pxeboot",
         "github.com/u-root/u-root/cmds/ip",
       },
    },
  },
  Archiver: uroot.CPIOArchiver,
  ...
}
if err := uroot.CreateInitramfs(opts); err != nil {
   ...
}
```

Reply:
This is fine but it's not fine. It's going to get increasingly inconvenient to use since it's not a Glob, and hence every use of this will need to change their code every time u-root gets a new command. I'm already finding this inconvenient on NiChrome. Pretty please, glob, pretty please?

Reply:
There already is a glob. You can specify packages like this:

```go
Packages: []string{
  "./cmds/[a-d]*",
},
```

Reply:
ping?

Reply:
sorry. Let me try that ... last time it did not seem to work. Well actually if you say it works, it works ...

Reply:
Relative file system path globs work, but not globs of package names. So, this will work:

```go
Packages: []string{
  "./cmds/[a-d]*",
  "github.com/u-root/u-root/cmds/ip",
},
```

but not:

```go
Packages: []string{
  "github.com/u-root/u-root/cmds/[a-d]*"
},
```

(making the latter work *well* would be a lot of work and isn't really worth it)